providers/env/env.go

```diff
@@ -12,9 +12,10 @@ import (
 
 // Env implements an environment variables provider.
 type Env struct {
-	prefix string
-	delim  string
-	cb     func(key string, value string) (string, interface{})
+	prefix  string
+	delim   string
+	environ []string
```

Comment:
Setting `environ` here means that any updates to the environment variables (namely via `os.Setenv`) won't be reflected when later reads are done. It may be better to specify this as a func:
```
environ func() []string
```

and then using `os.Environ` as the func by default .

Reply:
Makes sense, I adjusted the code to reflect the change